reference/matrix/csr_kernels.cpp

```diff
@@ -396,6 +397,80 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_CALCULATE_MAX_NNZ_PER_ROW_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void convert_to_hybrid(std::shared_ptr<const ReferenceExecutor> exec,
+                       matrix::Hybrid<ValueType, IndexType> *result,
+                       const matrix::Csr<ValueType, IndexType> *source)
+{
+    auto num_rows = result->get_size()[0];
+    auto num_cols = result->get_size()[1];
+    auto strategy = result->get_strategy();
+    auto ell_lim = strategy->get_ell_num_stored_elements_per_row();
+    auto coo_lim = strategy->get_coo_nnz();
+    auto coo_val = result->get_coo_values();
+    auto coo_col = result->get_coo_col_idxs();
+    auto coo_row = result->get_coo_row_idxs();
+
+    // Initial Hybrid Matrix
+    for (size_type i = 0; i < result->get_ell_num_stored_elements_per_row();
+         i++) {
+        for (size_type j = 0; j < result->get_ell_stride(); j++) {
+            result->ell_val_at(j, i) = zero<ValueType>();
+            result->ell_col_at(j, i) = 0;
+        }
+    }
+    for (size_type i = 0; i < result->get_coo_num_stored_elements(); i++) {
+        coo_val[i] = zero<ValueType>();
+        coo_col[i] = 0;
+        coo_row[i] = 0;
+    }
+
+    const auto csr_row_ptrs = source->get_const_row_ptrs();
+    const auto csr_vals = source->get_const_values();
+    size_type csr_idx = 0;
+    size_type coo_idx = 0;
+    for (IndexType row = 0; row < num_rows; row++) {
+        size_type ell_idx = 0;
+        while (csr_idx < csr_row_ptrs[row + 1]) {
+            const auto val = csr_vals[csr_idx];
+            if (val != zero<ValueType>()) {
```

Comment:
Is that test correct? To count `max_nnz` you also count zeros. Doesn't that mean you should also store explicit zeros here? In general, what is our policy on these issues? For some conversions, I think we preserve zeros (CSR <-> COO,CSR->SELLP) but some others that does not seem to be the case (ELL <-> CSR). The lists are non exhaustive.

Reply:
you are right. The result is not the expected hybrid matrix.